device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -403,7 +381,7 @@ public void sendReportedProperties(Set<Property> reportedProperties, Integer ver
      * @param callbackContext a context to be passed to the callback. Can be {@code null}.
      * @throws IllegalArgumentException if provided callback is null
      */
-    public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext) throws IllegalArgumentException
```

Comment:
Possibly a more controversial change, but "set" is a more commonly used verb than "register" for a callback like this